packages/contracts-bedrock/contracts/test/L1CrossDomainMessenger.t.sol

```diff
@@ -384,4 +385,302 @@ contract L1CrossDomainMessenger_Test is Messenger_Initializer {
         assertEq(L1Messenger.successfulMessages(hash), false);
         assertEq(L1Messenger.receivedMessages(hash), true);
     }
+
+    function test_relayMessage_legacy_succeeds() external {
+        address target = address(0xabcd);
+        address sender = Predeploys.L2_CROSS_DOMAIN_MESSENGER;
+
+        // Compute the message hash.
+        bytes32 hash = Hashing.hashCrossDomainMessageV1(
+            // Using a legacy nonce with version 0.
+            Encoding.encodeVersionedNonce(0, 0),
+            sender,
+            target,
+            0,
+            0,
+            hex"1111"
+        );
+
+        // Set the value of op.l2Sender() to be the L2 Cross Domain Messenger.
+        vm.store(address(op), bytes32(senderSlotIndex), bytes32(abi.encode(sender)));
+
+        // Target should be called with expected data.
+        vm.expectCall(target, hex"1111");
+
+        // Expect RelayedMessage event to be emitted.
+        vm.expectEmit(true, true, true, true);
+        emit RelayedMessage(hash);
+
+        // Relay the message.
+        vm.prank(address(op));
+        L1Messenger.relayMessage(
```

Comment:
We could update this call to use the object like syntax where its key value pairs, would remove the need to have `// value` and make it obvious what each arg is

Reply:
```solidity
L1Messenger.relayMessage({
  key: value,
  ...
})
```

Reply:
Let's do that as part of a separate PR, it would be a style change we should be consistent about